### ISSUE TITLE
Discrepancy between values of "rpc_success" and "rpc_duration_.*_success_count"?

### ISSUE THREAD

**Issue by Aracki** [ethereum/go-ethereum#30338]
#### System information

Geth version: `v1.14.7`

#### Expected behaviour

I'd expect these 2 numbers to give the same value.

#### Actual behaviour

For one specific Geth node, these two expressions should return the same value but often we see big discrepancy!
```
rate({__name__=~"rpc_success"}[5m])
sum(rate((label_replace({__name__=~"rpc_duration_.*_success_count"}, "method", "$1", "__name__", "rpc_duration_(.+)_success_count")[5m:])))
```

I've seen [this comment](https://github.com/ethereum/go-ethereum/blob/master/rpc/handler.go#L517-L518) about:
> 	// Collect the statistics for RPC calls if metrics is enabled.
	// We only care about pure rpc call. Filter out subscription.

Does that mean a lot of RPC methods are not recorded within `rpc_duration_.*_success_count` metric group? Not sure if this is a bug, or this is intended not to record all methods?

It would be nice if we can break down on how many each method is being called and not rely only on aggregated `rpc_success`.

**Comment by karalabe** [comment]
Ok, took me a while to figure your question out :D

So, `geth_rpc_success` tracks a global counter that's always updated from the start of Geth, going up.

On the other hand, the duration version is implemented with a histogram internally, which only aggregates the last N (1024) items and calculates running averages. It's a bit of a wonky invention unfortunately, we thought the lib we used was more meaningful but it turned out it just named it "histogram" for funsies but did not actually implement it as such.

As such, currently in our metrics we have a few "histograms" that are just a bit wonky, instead of being precise, they are doing some averaging over some ranges. We'd need to rethink this a bit, it was fairly low on our priority list as we only used it informatively to see general trend lines, not as very accurate numbers.
